platforms/core-runtime/launcher/src/main/java/org/gradle/launcher/cli/CommandLineActionCreator.java

```diff
@@ -35,8 +36,11 @@ public interface CommandLineActionCreator {
     /**
      * Creates an executable action from the given command-line args.
      *
+     * @param parser the parser that was used to parse the command-line args
+     * @param commandLine the parsed command-line args
+     * @param parameters a supplier for the parameters created from the command-line args
      * @return {@code null} if this creator does not know how to create an action from the given command-line args
      */
     @Nullable
-    Action<? super ExecutionListener> createAction(CommandLineParser parser, ParsedCommandLine commandLine);
+    Action<? super ExecutionListener> createAction(CommandLineParser parser, ParsedCommandLine commandLine, Supplier<Parameters> parameters);
```

Comment:
🤔 I don't quite follow why this is a Supplier? 

It looks like we always call `get()` immediately on it and we're creating the supplier just before the method is called.  Can't we just pass Parameters?  We don't need to memoize them because this is never reused.

Reply:
It's not always used everywhere, and I wanted to avoid introducing different behavior where it wasn't parsed before. It's probably not that expensive, so I could always build it.

Reply:
IIUC, the only place we don't use Parameters is when we're handling `-h`.  Everything else requires the Parameters to be created.  Is that right?

Reply:
Yeah. I'll change it.